pymongo/collection.py

```diff
@@ -2280,6 +2279,178 @@ def index_information(
             info[index.pop("name")] = index
         return info
 
+    def list_search_indexes(
+        self,
+        name: Optional[str] = None,
+        session: Optional["ClientSession"] = None,
+        comment: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> CommandCursor[_DocumentType]:
+        """Return a cursor over search indexes for the current collection.
+
+        :Parameters:
+            `name` (optional) - If given, the name of the index to search
+                for.  Only indexes with matching index names will be returned.
+                If not given, all search indexes for the current collection
+                will be returned.
+            - `session` (optional): a
+            :class:`~pymongo.client_session.ClientSession`.
+            - `comment` (optional): A user-provided comment to attach to this
+            command.
+
+        :Returns:
+          A :class:`~pymongo.command_cursor.CommandCursor` over the result
+          set.
+
+        .. note:: requires a MongoDB server version 7.0+ Atlas cluster.
+        """
+        if name is None:
+            pipeline: _Pipeline = [{"$listSearchIndexes": {}}]
+        else:
+            pipeline = [{"$listSearchIndexes": {"name": name}}]
+        return self.aggregate(pipeline, session, comment=comment, **kwargs)
+
+    def create_search_index(
+        self,
+        description: SearchIndexModel,
+        session: Optional["ClientSession"] = None,
+        comment: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> str:
+        """Create a single search index for the collection.
+
+        :Parameters:
+          - `description` - The index description for the new search index.
+          - `session` (optional): a
+            :class:`~pymongo.client_session.ClientSession`.
+          - `comment` (optional): A user-provided comment to attach to this
+            command.
+          - `**kwargs` (optional): optional arguments to the createSearchIndexes
+            command (like maxTimeMS) can be passed as keyword arguments.
+
+        :Returns:
+          The name of the new search index.
+
+        .. note:: requires a MongoDB server version 7.0+ Atlas cluster.
```

Comment:
Let's add `versionadded` tags to all the new apis.

Reply:
Done